src/main/java/gov/cabinetofice/gapuserservice/web/LoginControllerV2.java

```diff
@@ -98,21 +101,22 @@ public RedirectView login(final @RequestParam(name = REDIRECT_URL_NAME) Optional
     public RedirectView redirectAfterLogin(
             final @CookieValue(name = STATE_COOKIE) String stateCookie,
             final HttpServletResponse response,
-            final @RequestParam String code,
-            final @RequestParam String state
-    ) {
+             final @RequestParam String code,
+            final @RequestParam String state) {
         final JSONObject tokenResponse = oneLoginService.getOneLoginUserTokenResponse(code);
+        final String authToken = tokenResponse.getString("access_token");
+        final String tokenHint = tokenResponse.getString("id_token");
+
         IdTokenDto decodedIdToken = oneLoginService.getDecodedIdToken(tokenResponse);
```

Comment:
here you could modify `getDecodedIdToken` to instead be `decodeIdToken` which takes the idToken so it's not called `tokenResponse.getString("id_token")` twice - not necessary though

I would rename authToken to accessToken and tokenHint to idToken though for clarity

Reply:
Good shout, i'll update that